presto-main/src/main/java/io/prestosql/sql/planner/optimizations/PruneUnreferencedOutputs.java

```diff
@@ -258,6 +258,10 @@ public PlanNode visitJoin(JoinNode node, RewriteContext<Set<Symbol>> context)
         @Override
         public PlanNode visitSemiJoin(SemiJoinNode node, RewriteContext<Set<Symbol>> context)
         {
+            if (!context.get().contains(node.getSemiJoinOutput())) {
```

Comment:
consider adding test for this

Reply:
I added this behavior to make `PruneUnreferencedOutputs.visitSemiJoin()` consistent with the iterative pruning rules for SemiJoinNode. The visitor method is not tested, so that would be the only piece tested. On the other hand, the iterative rules are thoroughly tested (including this behavior), and soon this Optimizer will be replaced by them.